### PR TITLE
Add models and clients for helm release and feature resource

### PR DESCRIPTION
**Pull request by Shubbhang351**
1. **What this PR does / why we need it**:

Add models and clients for helm release and feature resource

### CODE REVIEW

File: internal/resources/helmfeature/scope/scope.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	FullnameCluster      *helmclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdHelmFullName
+	FullnameClusterGroup *helmclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupFluxcdHelmFullName
+}
+
+var (
+	ScopesAllowed = [...]string{commonscope.ClusterKey, commonscope.ClusterGroupKey}
+	ScopeSchema   = commonscope.GetScopeSchema(
+		commonscope.WithDescription(fmt.Sprintf("Scope for the helm release, having one of the valid scopes: %v.", strings.Join(ScopesAllowed[:], `, `))),
+		commonscope.WithScopes(ScopesAllowed[:]))
+)
```
Comment (ishangupta-ds) [inline internal/resources/helmfeature/scope/scope.go:32]:
```suggestion
		commonscope.WithDescription(fmt.Sprintf("Scope for the Helm feature, having one of the valid scopes: %v.", strings.Join(ScopesAllowed[:], `, `))),
		commonscope.WithScopes(ScopesAllowed[:]))
)
```

File: internal/resources/helmrelease/scope/scope.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	FullnameCluster      *releaseclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdHelmReleaseFullName
+	FullnameClusterGroup *releaseclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdHelmReleaseFullName
+}
+
+var (
+	ScopesAllowed = [...]string{commonscope.ClusterKey, commonscope.ClusterGroupKey}
+	ScopeSchema   = commonscope.GetScopeSchema(
+		commonscope.WithDescription(fmt.Sprintf("Scope for the helm release, having one of the valid scopes: %v.", strings.Join(ScopesAllowed[:], `, `))),
+		commonscope.WithScopes(ScopesAllowed[:]))
+)
```
Comment (ishangupta-ds) [inline internal/resources/helmrelease/scope/scope.go:32]:
```suggestion
		commonscope.WithDescription(fmt.Sprintf("Scope for the Helm release, having one of the valid scopes: %v.", strings.Join(ScopesAllowed[:], `, `))),
		commonscope.WithScopes(ScopesAllowed[:]))
)
```

File: internal/resources/helmrelease/spec/cluster_scope.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+// 	buf := bytes.NewBuffer(nil)
+// 	io.Copy(buf, inputFile)
+
+// 	_, err = yaml.Marshal(buf.String())
+// 	if err != nil {
+// 		return "", err
+// 	}
+
+// 	return buf.String(), nil
+// }
```
Comment (ishangupta-ds) [inline internal/resources/helmrelease/spec/cluster_scope.go:121]: Can we remove this if it is unused?

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

File: internal/resources/helmrelease/status/statue.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var StatusSchema = &schema.Schema{
+	Type:        schema.TypeMap,
+	Description: "Status for the Repository.",
+	Computed:    true,
+	Elem:        &schema.Schema{Type: schema.TypeString},
+}
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/helmrelease/status/statue.go]: updated

File: internal/resources/helmrelease/status/cluster_group_scope.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+	if status.Phase == nil {
+		return data
+	}
+
+	flattenStatusData := make(map[string]interface{})
+
+	flattenStatusData[phaseKey] = string(*status.Phase)
+
+	return flattenStatusData
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/helmrelease/status/cluster_group_scope.go:23]: but this we have decided in design doc to take phase for cluster_group only

File: internal/resources/helmrelease/status/statue.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/helmrelease/status/statue.go]: UPdated

File: internal/resources/helmrelease/spec/cluster_scope.go [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	flattenSpecData[InlineConfigKey] = spec.InlineConfiguration
+	flattenSpecData[IntervalKey] = spec.Interval
+
+	var chartRef = make(map[string]interface{})
+
+	log.Println("<><> type of repo")
+	log.Println("<><> type of repo")
+	log.Println("<><> type of repo >> is is helm", *spec.ChartRef.RepositoryType == *releaseclustermodel.NewVmwareTanzuManageV1alpha1ClusterNamespaceFluxcdHelmReleaseRepositoryType(releaseclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdHelmReleaseRepositoryTypeHELM))
+	log.Println("<><> type of repo >> is is git", *spec.ChartRef.RepositoryType == *releaseclustermodel.NewVmwareTanzuManageV1alpha1ClusterNamespaceFluxcdHelmReleaseRepositoryType(releaseclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdHelmReleaseRepositoryTypeGIT))
+
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/helmrelease/spec/cluster_scope.go]: updated

File: internal/resources/helmrelease/spec/cluster_scope.go (hunk 3 above)
Comment (Shubbhang351) [reply internal/resources/helmrelease/spec/cluster_scope.go]: updated

**Review by ishangupta-ds — APPROVED** [review]
LGTM!

File: internal/client/http_client.go [hunk 7]
```diff
@@ -112,6 +116,10 @@ func newHTTPClient(httpClient *transport.Client) *TanzuMissionControl {
 		ClusterTanzuPackageService:                    tanzupackageclusterclient.New(httpClient),
 		TanzupackageResourceService:                   packageclusterclient.New(httpClient),
 		PackageInstallResourceService:                 pkginstallclusterclient.New(httpClient),
+		ClusterHelmReleaseResourceService:             helmreleaseclusterclient.New(httpClient),
+		ClusterGroupHelmReleaseResourceService:        helmreleaseclustergroupclient.New(httpClient),
+		ClustreHelmResourceService:                    helmfeatureclusterclient.New(httpClient),
```
Comment (ramya-bangera) [inline internal/client/http_client.go:121]: nit: typo `ClustreHelmResourceService -> ClusterHelmResourceService`

File: internal/client/http_client.go [hunk 8]
```diff
@@ -112,6 +116,10 @@ func newHTTPClient(httpClient *transport.Client) *TanzuMissionControl {
...
+		ClustreGroupHelmResourceService:               helmfeatureclustergroupclient.New(httpClient),
```
Comment (ramya-bangera) [inline internal/client/http_client.go:122]: nit: typo `ClustreGroupHelmResourceService -> ClusterGroupHelmResourceService`

File: internal/resources/helmfeature/helm_feature_resource.go [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+	commonscope "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common/scope"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/helmfeature/scope"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/helmfeature/status"
+)
+
+const (
+	ResourceName = "tanzu-mission-control_helm_feature"
```
Comment (ramya-bangera) [inline internal/resources/helmfeature/helm_feature_resource.go:18]: nit: newline afterL18 for better readability

**Review by ramya-bangera — COMMENTED** [review]
LGTM

File: internal/client/http_client.go (hunk 7 above)
Comment (Shubbhang351) [reply internal/client/http_client.go]: Updated

File: internal/client/http_client.go (hunk 8 above)
Comment (Shubbhang351) [reply internal/client/http_client.go]: Updated

File: internal/resources/helmfeature/helm_feature_resource.go (hunk 9 above)
Comment (Shubbhang351) [reply internal/resources/helmfeature/helm_feature_resource.go:18]: Updated

**Review by ishangupta-ds — APPROVED** [review]
LGTM!

**Review by ramya-bangera — APPROVED** [review]

**Pull request merged**